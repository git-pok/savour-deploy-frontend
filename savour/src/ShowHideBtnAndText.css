.ShowHideBtnAndText-p {
    display: block;
    margin: 5px auto 5px auto;
    text-align: left;
    color: rgb(255, 255, 255);
}

.ShowHideBtnAndText-button {
    background-color: rgb(223, 216, 85);
    color: rgb(0, 0, 0);
    border: 3px inset rgb(223, 216, 85);
    transition: .5s ease-in-out background-color;
}

.ShowHideBtnAndText-button:hover {
    background-color: rgb(255, 255, 255);
    color: rgb(0, 0, 0);
    border: 3px inset rgb(255, 255, 255);
    transition: .5s ease-in-out background-color;
}