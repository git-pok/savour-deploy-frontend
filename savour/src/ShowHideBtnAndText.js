import useToggleState from './hooks/useToggleState.js';
import './ShowHideBtnAndText.css';

/**
 * ShowHideBtnAndText
 * Props: paragraphArr
 *    paragraphArr: array of nested objects. Objects
 *    have text prop =>
 *    [{ text: "Prep Time: 20mins" }]
 * Renders: div with paragraphs, and button
 *    that shows/hides them on click.
*/
const ShowHideBtnAndText = ({paragraphArr}) => {
  const [ isClicked, setIsClicked ] = useToggleState(false);
  const click = () => {
    setIsClicked();
  }
  return (
    <>
    <button className="ShowHideBtnAndText-button" onClick={click}>SEE MORE</button>
      {
        paragraphArr && paragraphArr.map((val, idx) => (
          isClicked && <p key={idx} className="ShowHideBtnAndText-p">{val.text}</p>
        ))
      }
    </>
  );
}

export default ShowHideBtnAndText;