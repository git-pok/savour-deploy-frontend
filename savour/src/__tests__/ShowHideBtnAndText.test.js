import ShowHideBtnAndText from '../ShowHideBtnAndText.js';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

test("renders ShowHideBtnAndText without crashing", () => {
  render(
  <MemoryRouter>
    <ShowHideBtnAndText
      paragraphArr={
        [
          {
            text: "Test"
          }
        ]
    } />
  </MemoryRouter>);
});

test("renders snapshot without crashing", () => {
  const { asFragment } = render(
    <MemoryRouter>
    <ShowHideBtnAndText
      paragraphArr={
        [
          {
            text: "Test"
          }
        ]
    } />
    </MemoryRouter>
  );

  expect(asFragment()).toMatchSnapshot();
});