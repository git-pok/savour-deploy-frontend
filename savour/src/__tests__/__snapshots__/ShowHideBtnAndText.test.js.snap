// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders snapshot without crashing 1`] = `
<DocumentFragment>
  <button
    class="ShowHideBtnAndText-button"
  >
    SEE MORE
  </button>
</DocumentFragment>
`;
